### ISSUE TITLE
Sources get sha256 appended in file name

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/rattler-build#402]
```yml
# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
package:
  name: test-package
  version: 0.1.0
source:
  - url: https://repo1.maven.org/maven2/net/java/dev/jna/jna/5.8.0/jna-5.8.0.jar
    sha256: 930273cc1c492f25661ea62413a6da3fd7f6e01bf1c4dcc0817fc8696a7b07ac
    # file_name: jna-5.8.0.jar

build:
  script:
    - ls -lah
    - ls -lah | grep jna-5.8.0.jar
  number: 0
```

```
$ rattler-build build --recipe conda-recipe/recipe.yaml
...
+ ls -lah
total 3408
drwxr-xr-x  5 pavel  staff   160B Dec  6 13:36 .
drwxr-xr-x  5 pavel  staff   160B Dec  6 13:36 ..
-rw-r--r--  1 pavel  staff   6.9K Dec  6 13:36 build_env.sh
-rw-r--r--  1 pavel  staff   300B Dec  6 13:36 conda_build.sh
-rw-r--r--  1 pavel  staff   1.6M Nov 29 12:18 jna-5.8.0_930273cc.jar
+ ls -lah
+ grep snowflake-jdbc-3.13.27.jar
Error:   × Build failed
```

Using `rattler-build 0.6.0`.
When I uncomment the `file_name: jna-5.8.0.jar`, it works.

**Comment by wolfv** [comment]
ouch!

**Comment by wolfv** [comment]
This is fixed in #412
